sql/core/src/test/scala/org/apache/spark/sql/StatisticsColumnSuite.scala

```diff
@@ -0,0 +1,355 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.sql.{Date, Timestamp}
+
+import org.apache.spark.sql.catalyst.{InternalRow, TableIdentifier}
+import org.apache.spark.sql.catalyst.parser.ParseException
+import org.apache.spark.sql.catalyst.plans.logical.ColumnStat
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+import org.apache.spark.sql.execution.command.AnalyzeColumnCommand
+import org.apache.spark.sql.test.SQLTestData.ArrayData
+import org.apache.spark.sql.types._
+
+class StatisticsColumnSuite extends StatisticsTest {
+  import testImplicits._
+
+  test("parse analyze column commands") {
+    def assertAnalyzeColumnCommand(analyzeCommand: String, c: Class[_]) {
+      val parsed = spark.sessionState.sqlParser.parsePlan(analyzeCommand)
+      val operators = parsed.collect {
+        case a: AnalyzeColumnCommand => a
+        case o => o
+      }
+      assert(operators.size == 1)
+      if (operators.head.getClass != c) {
+        fail(
+          s"""$analyzeCommand expected command: $c, but got ${operators.head}
+             |parsed command:
+             |$parsed
+           """.stripMargin)
+      }
+    }
+
+    val table = "table"
+    assertAnalyzeColumnCommand(
+      s"ANALYZE TABLE $table COMPUTE STATISTICS FOR COLUMNS key, value",
+      classOf[AnalyzeColumnCommand])
+
+    intercept[ParseException] {
+      sql(s"ANALYZE TABLE $table COMPUTE STATISTICS FOR COLUMNS")
+    }
+  }
+
+  test("analyzing columns of non-atomic types is not supported") {
+    val tableName = "tbl"
+    withTable(tableName) {
+      Seq(ArrayData(Seq(1, 2, 3), Seq(Seq(1, 2, 3)))).toDF().write.saveAsTable(tableName)
+      val err = intercept[AnalysisException] {
+        sql(s"ANALYZE TABLE $tableName COMPUTE STATISTICS FOR COLUMNS data")
+      }
+      assert(err.message.contains("Analyzing columns is not supported"))
+    }
+  }
+
+  test("check correctness of columns") {
+    val table = "tbl"
+    val colName1 = "abc"
+    val colName2 = "x.yz"
+    val quotedColName2 = s"`$colName2`"
```

Comment:
you can inline this variable, i.e.`CREATE TABLE $table ($colName1 int, `$colName2` string)`